rfcs/0035-create-connection.md

```diff
@@ -0,0 +1,143 @@
+
+```sql
+CREATE CONNECTION demo_connection_1 TO KAFKA
+  (PROPERTIES.BOOTSTRAP.SERVER = 'broker1:9092,broker2:9092');
+
+CREATE CONNECTION demo_connection_2 TO KAFKA
+  (PROPERTIES.BOOTSTRAP.SERVER.PRIVATE_LINK = ['broker1:9092' USING AWS_PRIVATE_LINK {{ connection_name }} (PORT 9092), 'broker2:9092' USING AWS_PRIVATE_LINK {{ connection_name }} (PORT 9092)]];
+```
+
+For general usage, the following fields are allowed:
+
+|Field|Value|Required|Description|
+|--|--|--|--|
+|`PROPERTIES.BOOTSTRAP.SERVER`| varchar | Conditional | The Kafka bootstrap server address. |
+|`PROPERTIES.BOOTSTRAP.SERVER.PRIVATE_LINK`| varchar[] | Conditional | The Kafka bootstrap server address with AWS Private Link settings. This field is not allowed to set at the same time as field `PROPERTIES.BOOTSTRAP.SERVER`. |
+
+For security related fields (SSL/SASL), the fields are identical to [this doc](https://materialize.com/docs/sql/create-connection/#kafka-ssl).
+
+**Notice**: Risingwave will not check the connection to Kafka is valid or not.
+
+> Some notes for implementation
+>
+> 1. Implement libkafka's DNS resolve callback
+> 2. Store the dns mapping from the user's brokers address to private link endpoint ip
+> 3. Interface for creating and modifying dns mappings
+
+#### Kinesis
+
+A Kinesis connection establishes a link to a Kinesis stream. Users can use Kinesis connections to create Kinesis sources and sinks. The following fields are required to create a Kinesis connection:
+
+```sql
+CREATE CONNECTION demo_connection TO KINESIS
+  (aws.region='user_test_topic',
+   endpoint='172.10.1.1:9090,172.10.1.2:9090',
+   aws.credentials.role.arn='arn:aws-cn:iam::602389639824:role/demo_role',
+   aws.credentials.role.external_id='demo_external_id',
+   ...
+  );
+```
+
+Accepted fields are listed below, which is a subset of [the doc](https://www.risingwave.dev/docs/current/create-source-kinesis/#parameters).
+
+|Field|Required|Description|
+|--|--|--|
+|`aws.region`| yes | AWS service region. For example, US East (N. Virginia). |
+|`endpoint`| Optional |URL of the entry point for the AWS Kinesis service.|
+|`aws.credentials.access_key_id`| Conditional | This field indicates the access key ID of AWS. It must appear in pairs with aws.credentials.secret_access_key.|
+|`aws.credentials.secret_access_key`|Conditional | This field indicates the secret access key of AWS. It must appear in pairs with aws.credentials.access_key_id.|
+|`aws.credentials.session_token`|Optional| The session token associated with the temporary security credentials.|
+|`aws.credentials.role.arn`|Optional| The Amazon Resource Name (ARN) of the role to assume.|
+|`aws.credentials.role.external_id`|Optional|The [external id](https://aws.amazon.com/blogs/security/how-to-use-external-id-when-granting-access-to-your-aws-resources/) used to authorize access to third-party resources.|
+
+**Notice**: Risingwave will not check the connection to Kinesis is valid or not.
```

Comment:
Reason?

Reply:
in `create connection`, we don't specify topic/stream. We may don't have access to list topics globally and there is no proper API for the validation.

Reply:
adding a default topic to do the validation is acceptable but has concerns about misuse.